features/lorawan/lorastack/phy/LoRaPHY.cpp

```diff
@@ -443,20 +444,20 @@ uint8_t LoRaPHY::get_bandwidth(uint8_t dr)
 }
 
 uint8_t LoRaPHY::enabled_channel_count(bool joined, uint8_t datarate,
-                                       uint16_t *mask_list,
-                                       uint8_t mask_list_size,
+                                       uint16_t *channel_mask,
+                                       uint8_t mask_size,
                                        uint8_t *channel_indices,
                                        uint8_t *delayTx)
 {
     uint8_t count = 0;
     uint8_t delay_transmission = 0;
 
-    for (uint8_t i = 0, k = 0; i < phy_params.max_channel_cnt && k < mask_list_size;
+    for (uint8_t i = 0, k = 0; i < phy_params.max_channel_cnt && k < mask_size;
          i += CHANNELS_IN_MASK, k++) {
 
         for (uint8_t j = 0; j < CHANNELS_IN_MASK; j++) {
 
-            if ((mask_list[k] & (1 << j)) != 0) {
+            if (MASK_BIT_TEST(channel_mask[k], j) != 0) {
```

Comment:
I was thinking this would work a like `MASK_BIT_TEST(channel_mask, channel_number)`, which would definitely simplify this routine, avoiding the need for a double loop.

    // replace mask_size with channel_cnt
    if (channel_cnt > phy_params.max_channel_cnt) {
        channel_cnt = phy_params.max_channel_cnt;
    }

    for (int channel = 0; channel < channel_cnt; channel++) {
        if (MASK_BIT_TEST(channel_mask, channel)) {
            .. // replace all i+j with channel
        }
    }

Part of the point of the macro is to try to avoid having to think/worry about the container size for the mask. Simpler to always just think of it as an array of max_channel bits.

Reply:
We don't know the channel_cnt at this moment. That is the purpose of this routine to count number of usable channels and give us their indices.

Reply:
However, I got your point.

Reply:
You're right, it should be `max_channels`.